pages/learn/deploy/delta.md

```diff
@@ -8,30 +8,40 @@ thumbnail: /img/runtime/ResinSupervisorDelta.png
+
 ## Delta behavior
 
-If you are using delta updates, you might notice the following changes in {{ $names.company.lower }} behavior:
+Before devices can update using deltas, a delta must be generated between the release a device is currently running and the one it is updating to. There are two ways deltas are generated:
+
+1. Automatically between the last successful and the new release during a build on the {{$names.cloud.lower}} builder - these are known as "build-time" deltas.
+2. Automatically when a device is requesting to update from or to a release for which no delta has been generated before - these are known as "on-demand" deltas and can typically happen if the device was not on the latest release when the new release was pushed, or the user did not use the {{$names.cloud.lower}} builder for the new release.
+
+__Note:__ Delta updates that were enabled for applications created prior to December 2019 (v2) do not generate build-time deltas, and will always generate an on-demand delta.
 
-The *Download progress* bar on the dashboard might show for only a short time—much shorter than in a normal application update. This is because in the most common development patterns, there are usually very small changes between one version of the application image and the next (e.g. fixing typos, adding a new source file, or installing an extra OS package), and when using deltas these changes are downloaded much quicker than before.
+On-demand deltas may take a while to generate, depending primarily on the size of the images involved, and while doing so, the device keeps polling the API for its readiness. If a delta exists, this step is skipped.
 
-Delta updates are resumable, so if the connection drops or otherwise stalls, the update will resume from the last byte received. A user can configure the number of times a delta is retried before it bails out and signals failure to the supervisor. This is set by the `RESIN_SUPERVISOR_DELTA_RETRY_COUNT` configuration variable.
+The *Download progress* bar on the dashboard might show for only a short time, much shorter than in a normal application update. In the most common development patterns, there are usually very small changes between one version of the application image and the next (e.g. fixing typos, adding a new source file, or installing an extra OS package), so when using deltas these changes are downloaded much quicker than before.
 
-In addition, the wait time before a connection is considered stalled and the time between retries are configurable, using `RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT` and `RESIN_SUPERVISOR_DELTA_RETRY_INTERVAL`. These are both specified in milliseconds. The defaults for these options allow for a 20 minute time frame where no byte is received before giving up, after which the supervisor retries the delta from the beginning.
+When the update finishes, the previous services are stopped and the new services are started and the prior image removed.
```

Comment:
That is accurate for the default case, but depends on the specific update strategy used. I think it's worth mentioning that and also linking the relevant docs here.